testing/test_config.py

```diff
@@ -1273,6 +1273,191 @@ def test_with_existing_file_in_subdir(self, tmpdir) -> None:
             assert inifile is None
 
 
+class TestAppendIniArgs:
+    def test_append_ini_usage_error_bad_style(self, testdir):
```

Comment:
If we follow my suggestion of using standalone functions, almost all of those tests can be changed into lightning fast unittests instead, instead of full integration tests. We can then only have a single integration test to ensure everything is working correctly.

Reply:
@nicoddemus agreed. I'll remove the class I made since we already test these things in [TestOverrideIniArgs](https://github.com/pytest-dev/pytest/blob/master/testing/test_config.py#L1277).

I think we can get away with two tests:
- One test parametrized to test valid overrides
- One test parametrized to test invalid overrides

Reply:
I spoke too soon. I'd like some more clarity on what you're thinking, because I think I'm misunderstanding you